docs/defining-data-types.md

```diff
@@ -3,48 +3,64 @@
 It is possible to easily define complex data types using the `type` keyword.
 
 ```py
-# A Boolean is either True or False
-type Bool = True | False
+#{
+  A Boolean is either True or False 
+}#
+type Bool:
+  True
+  False
 ```
 
 If a constructor has any arguments, parentheses are necessary around it:
 ```py
-# An option either contains some value, or None
-type Option = (Some val) | None
+#{
+An option either contains some value, or None
+}#
+type Option:
+ Some { value }
+ None
 ```
 
-If the data type has a single constructor, it can be destructured using `let`:
-```py
-# A Box is a wrapper around a value.
-type Boxed = (Box val)
 
-let (Box value) = boxed; value
-```
 
 The fields of the constructor that is being destructured with the `match` are bound to the matched variable plus `.` and the field names.
 ```py
-Option.map = λoption λf
-  match option {
-    Some: (Some (f option.val))
-    None: None
-  }
+opt = Option/Some(1)
+match opt:
+  case Option/Some:
+    return opt.value
+  case Option/None:
+    return 0
 ```
 
 Rules can also have patterns.
 They work like match expressions with explicit bindings:
 
 ```py
-(Option.map (Some value) f) = (Some (f value))
-(Option.map None f) = None
+def Option/map(opt: Option(T), f: T -> U) -> Option(U):
```

Comment:
The point of this section is showing pattern matching functions, you can't just change to the other syntax